src/sentry/uptime/models.py

```diff
@@ -24,7 +24,7 @@
 from sentry.models.organization import Organization
 from sentry.remote_subscriptions.models import BaseRemoteSubscription
 from sentry.types.actor import Actor
-from sentry.uptime.types import DATA_SOURCE_UPTIME_SUBSCRIPTION
+from sentry.uptime.types import DATA_SOURCE_UPTIME_SUBSCRIPTION, ProjectUptimeSubscriptionMode
```

Comment:
just curious - how is tracking the `DATA_SOURCE_UPTIME_SUBSCRIPTION` type working for you? is that abstraction feeling weird at all?

(I was thinking we could wrap the process_data_source to inspect the type coming in and have this as an optional override)

Reply:
It's not a problem so far. But I suppose we could auto detect it based on a registry of data packet -> data source. Doesn't seem too high priority, but could be a nice thing to add later?